content/intro-to-storybook/react-native/en/composite-component.md

```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
-
-![Jest logo](/intro-to-storybook/logo-jest.png)
-
-### Unit tests with Jest
-
-Storybook stories paired with manual visual tests and snapshot tests (see above) go a long way to avoiding UI bugs. If stories cover a wide variety of component use cases, and we use tools that ensure a human checks any change to the story, errors are much less likely.
-
-However, sometimes the devil is in the details. A test framework that is explicit about those details is needed. Which brings us to unit tests.
-
-In our case, we want our `TaskList` to render any pinned tasks **before** unpinned tasks that it has passed in the `tasks` prop. Although we have a story (`withPinnedTasks`) to test this exact scenario, it can be ambiguous to a human reviewer that if the component **stops** ordering the tasks like this, it is a bug. It certainly won’t scream **“Wrong!”** to the casual eye.
+These changes result in the following UI:
 
-So, to avoid this problem, we can use Jest to render the story to the DOM and run some DOM querying code to verify salient features of the output.
+![TaskList with loading state](/intro-to-storybook/react-native-tasklist-loading.gif)
 
-Create a test file called `components/__tests__/TaskList.test.js`. Here, we’ll build out our tests that make assertions about the output.
-
-```javascript
-// components/__tests__/TaskList.test.js
-import * as React from 'react';
-import { create } from 'react-test-renderer';
-import TaskList from '../TaskList';
-import { withPinnedTasks } from '../TaskList.stories';
-import Task from '../Task';
-describe('TaskList', () => {
-  it('renders pinned tasks at the start of the list', () => {
-    const events = { onPinTask: jest.fn(), onArchiveTask: jest.fn() };
-    const tree = create(<TaskList tasks={withPinnedTasks} {...events} />);
-    const rootElement = tree.root;
-    const listofTasks = rootElement.findAllByType(Task);
-    expect(listofTasks[0].props.task.title).toBe('Task 6 (pinned)');
-  });
-});
-```
-
-![TaskList test runner](/intro-to-storybook/tasklist-testrunner.png)
-
-Note that we’ve been able to reuse the `withPinnedTasks` list of tasks in both story and unit test; in this way we can continue to leverage an existing resource (the examples that represent interesting configurations of a component) in many ways.
-
-Notice as well that this test is quite brittle. It's possible that as the project matures, and the exact implementation of the `Task` changes --perhaps using a different styling prop or a `Text` rather than an `TextInput`--the test will fail, and need to be updated. This is not necessarily a problem, but rather an indication to be careful about liberally using unit tests for UI. They're not easy to maintain. Instead rely on visual, snapshot tests where possible.
+Note the position of the pinned item in the list. We want the pinned item to render at the top of the list to make it a priority for our users.
```

Comment:
@dannyhw to avoid having the chapter end like this, as it seems rather abrupt, we can rephrase it to something like this:
"Success! We accomplished what we set out to do. If we check our updated UI, we can quickly see that our pinned tasks are now featured at the top of the list, matching the intended design. In the next chapters, we'll expand on what we've learned and continue adding complexity to our application by applying these principles to more complex UIs."

End with the GitHub aside mentioning to don't forget to commit the changes

Reply:
updated